SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -472,3 +345,176 @@ void USpatialWorkerConnection::OnConnectionFailure()
+	check(OpsProcessingThread);
+}
+
+void USpatialWorkerConnection::QueueLatestOpList()
+{
+	OpListQueue.Enqueue(Worker_Connection_GetOpList(WorkerConnection, 0));
+}
+
+void USpatialWorkerConnection::ProcessOutgoingMessages()
+{
+	while (!OutgoingMessagesQueue.IsEmpty())
+	{
+		TUniquePtr<FOutgoingMessage> OutgoingMessage;
+		OutgoingMessagesQueue.Dequeue(OutgoingMessage);
+
+		switch (OutgoingMessage->Type)
+		{
+		case EOutgoingMessageType::ReserveEntityIdsRequest:
+		{
+			FReserveEntityIdsRequest* Message = static_cast<FReserveEntityIdsRequest*>(OutgoingMessage.Get());
+
+			Worker_Connection_SendReserveEntityIdsRequest(WorkerConnection,
+				Message->NumOfEntities,
+				nullptr);
+			break;
+		}
+		case EOutgoingMessageType::CreateEntityRequest:
+		{
+			FCreateEntityRequest* Message = static_cast<FCreateEntityRequest*>(OutgoingMessage.Get());
+
+			Worker_Connection_SendCreateEntityRequest(WorkerConnection,
+				Message->Components.Num(),
+				Message->Components.GetData(),
+				Message->EntityId.IsSet() ? &(Message->EntityId.GetValue()) : nullptr,
+				nullptr);
+			break;
+		}
+		case EOutgoingMessageType::DeleteEntityRequest:
+		{
+			FDeleteEntityRequest* Message = static_cast<FDeleteEntityRequest*>(OutgoingMessage.Get());
+
+			Worker_Connection_SendDeleteEntityRequest(WorkerConnection,
+				Message->EntityId,
+				nullptr);
+			break;
+		}
+		case EOutgoingMessageType::ComponentUpdate:
+		{
+			FComponentUpdate* Message = static_cast<FComponentUpdate*>(OutgoingMessage.Get());
+
+			static const Worker_Alpha_UpdateParameters DisableLoopback{ false /* loopback */ };
+			Worker_Alpha_Connection_SendComponentUpdate(WorkerConnection,
+				Message->EntityId,
+				&Message->Update,
+				&DisableLoopback);
+			break;
+		}
+		case EOutgoingMessageType::CommandRequest:
+		{
+			FCommandRequest* Message = static_cast<FCommandRequest*>(OutgoingMessage.Get());
+
+			static const Worker_CommandParameters DefaultCommandParams{};
+			Worker_Connection_SendCommandRequest(WorkerConnection,
+				Message->EntityId,
+				&Message->Request,
+				Message->CommandId,
+				nullptr,
+				&DefaultCommandParams);
+			break;
+		}
+		case EOutgoingMessageType::CommandResponse:
+		{
+			FCommandResponse* Message = static_cast<FCommandResponse*>(OutgoingMessage.Get());
+
+			Worker_Connection_SendCommandResponse(WorkerConnection,
+				Message->RequestId,
+				&Message->Response);
+			break;
+		}
+		case EOutgoingMessageType::CommandFailure:
+		{
+			FCommandFailure* Message = static_cast<FCommandFailure*>(OutgoingMessage.Get());
+
+			Worker_Connection_SendCommandFailure(WorkerConnection,
+				Message->RequestId,
+				TCHAR_TO_UTF8(*Message->Message));
+			break;
+		}
+		case EOutgoingMessageType::LogMessage:
+		{
+			FLogMessage* Message = static_cast<FLogMessage*>(OutgoingMessage.Get());
+
+			Worker_LogMessage LogMessage{};
+			LogMessage.level = Message->Level;
+			LogMessage.logger_name = TCHAR_TO_UTF8(*Message->LoggerName.ToString());
```

Comment:
`TCHAR_TO_UTF8` is a temporary, passing its result to C API like this is undefined.
You should have `FTCHARToUTF8` as a local var instead (see `ConnectToLocator` for reference).